api/idp.go

```diff
@@ -0,0 +1,59 @@
+package api
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+// IdentityProviderType is the type of an identity provider.
+type IdentityProviderType string
+
+const (
+	// OAuth2IdentityProvider is the identity provider type for OAuth2.
+	OAuth2IdentityProvider IdentityProviderType = "OAUTH2"
+	// OIDCIdentityProvider is the identity provider type for OIDC.
+	OIDCIdentityProvider IdentityProviderType = "OIDC"
+)
+
+// FieldMapping uses for mapping user info from identity provider to Bytebase.
+type FieldMapping struct {
+	Identifier string `json:"identifier"`
+	Name       string `json:"name"`
+	Email      string `json:"email"`
+}
+
+// OAuth2IdentityProviderConfig is the structure for OAuth2 identity provider config.
+type OAuth2IdentityProviderConfig struct {
```

Comment:
Use proto for this. In the store package.

Reply:
Removed in this PR, and I'd like to add this in next PRs.

Reply:
SG